packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -74,6 +74,243 @@ class PlatformViewsService {
+  /// See Android's [MotionEvent.PointerCoords#size](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#size).
+  final double size;
+
+  /// See Android's [MotionEvent.PointerCoords#toolMajor](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#toolMajor).
+  final double toolMajor;
+
+  /// See Android's [MotionEvent.PointerCoords#toolMinor](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#toolMinor).
+  final double toolMinor;
+
+  /// See Android's [MotionEvent.PointerCoords#touchMajor](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#touchMajor).
+  final double touchMajor;
+
+  /// See Android's [MotionEvent.PointerCoords#touchMinor](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#touchMinor).
+  final double touchMinor;
+
+  /// The X component of the pointer movement.
+  ///
+  /// See Android's [MotionEvent.PointerCoords#x](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#x).
+  final double x;
+
+  /// The Y component of the pointer movement.
+  ///
+  /// See Android's [MotionEvent.PointerCoords#y](https://developer.android.com/reference/android/view/MotionEvent.PointerCoords.html#y).
+  final double y;
+
+  List<double> _asList() {
+    return <double>[
+      orientation,
+      pressure,
+      size,
+      toolMajor,
+      toolMinor,
+      touchMajor,
+      touchMinor,
+      x,
+      y,
+    ];
+  }
+
+  @override
+  String toString() {
+    return 'AndroidPointerCoords(orientation: $orientation, pressure: $pressure, size: $size, toolMajor: $toolMajor, toolMinor: $toolMinor, touchMajor: $touchMajor, touchMinor: $touchMinor, x: $x, y: $y)';
+  }
+}
+
+/// A Dart version of Android's [MotionEvent](https://developer.android.com/reference/android/view/MotionEvent).
+class AndroidMotionEvent {
+  /// Creates an AndroidMotionEvent.
+  ///
+  /// All parameters must not be null.
+  const AndroidMotionEvent({
+    @required this.downTime,
+    @required this.eventTime,
+    @required this.action,
+    @required this.pointerCount,
+    @required this.pointerProperties,
+    @required this.pointerCoords,
+    @required this.metaState,
+    @required this.buttonState,
+    @required this.xPrecision,
+    @required this.yPrecision,
+    @required this.deviceId,
+    @required this.edgeFlags,
+    @required this.source,
+    @required this.flags
+  }) : assert(downTime != null),
+       assert(eventTime != null),
+       assert(action != null),
+       assert(pointerCount != null),
+       assert(pointerProperties != null),
+       assert(pointerCoords != null),
+       assert(metaState != null),
+       assert(buttonState != null),
+       assert(xPrecision != null),
+       assert(yPrecision != null),
+       assert(deviceId != null),
+       assert(edgeFlags != null),
+       assert(source != null),
+       assert(flags != null);
```

Comment:
maybe assert that pointerProperties and pointerCoords are the same thing? (and match pointerCount, if we must have that explicitly there too)

Reply:
done. (though this required making the constructor non const)